library/Requests.php

```diff
@@ -821,7 +821,9 @@ public static function flattern($array) {
 	 * @return string Decompressed string
 	 */
 	public static function decompress($data) {
-		if (substr($data, 0, 2) !== "\x1f\x8b" && substr($data, 0, 2) !== "\x78\x9c") {
+		// All valid deflate, gzip header magic markers
+		$valid_magic = array("\x1f\x8b", "\x78\x01", "\x78\x5e", "\x78\x9c", "\x78\xda");
+		if (!in_array(substr($data, 0, 2), $valid_magic)) {
```

Comment:
This needs to use the `$strict` parameter to `in_array`, but otherwise looks good.

Reply:
Fixed in follow up commit. Tests included.